src/DynamoCoreWpf/ViewModels/PackageManager/PublishPackageViewModel.cs

```diff
@@ -2649,6 +2649,21 @@ private void PreviewPackageBuild()
                 return;
 
             var files = GetAllFiles().ToList();
+            if (!RetainFolderStructureOverride)
+            {
+                //Look for duplicate filenames to alert user
+                var duplicateFiles = files.GroupBy(x => Path.GetFileName(x))
+                    .Where(x => x.Count() > 1)
+                    .ToList();
+                if (duplicateFiles.Count() > 0)
+                {
+                    if (!DynamoModel.IsTestMode)
+                    {
+                        MessageBoxService.Show(System.Windows.Application.Current?.MainWindow, string.Format("{0} Duplicate file(s) found.\nFiles with same name will be overwritten in the final package, to avoid it, either rename or discard the duplicate files or enable Retain Folder Structure option.", duplicateFiles.Count()), "Duplicate Files", MessageBoxButton.OK, MessageBoxImage.Warning);
```

Comment:
We will need this message localized

Reply:
Yes, I mentioned that in the description.